packages/frontend/src/components/AudioRecorder/AudioRecorder.tsx

```diff
@@ -0,0 +1,161 @@
+  useEffect(() => {
+    const timer = window.setInterval(() => {
+      setRecordingTime(prevTime => prevTime + 1)
+    }, 1000)
+    return () => window.clearInterval(timer)
+  }, [])
+
+  const minutes = Math.floor(recordingTime / 60)
+    .toString()
+    .padStart(2, '0')
+  const seconds = (recordingTime % 60).toString().padStart(2, '0')
+
+  return (
+    <div>
+      <p>
+        {minutes}:{seconds}
+      </p>
+    </div>
+  )
+}
+
+/**
+ * Show a volume meter by stacking 3 div layers
+ * 1. a mask to show a grid of steps
+ * 2. a level layer to show/hide the coloured background
+ * 3. a coloured background with a color gradient from green to red
+ */
+const VolumeMeter = (prop: { volume: number }) => {
+  const steps = 10
+  const volumeBarRef = useRef<HTMLDivElement>(null)
+  const totalWidth = Number(volumeBarRef.current?.clientWidth) || 0
+  // doubling the volume shows a more realistic volume level
+  const level = Math.min(totalWidth * prop.volume * 2, totalWidth)
+  const levelWidth = totalWidth > 0 ? `${totalWidth - level}px` : '100%'
+  console.debug(`volume: ${(prop.volume * 100).toFixed(2)}%`)
+  return (
+    <div ref={volumeBarRef} className={styles.volumeBar}>
+      <div className={styles.mask}>
+        {Array.from({ length: steps }).map((_, index) => (
+          <div key={index} className={styles.step} />
+        ))}
+      </div>
+      <div className={styles.level} style={{ width: `${levelWidth}` }} />
+      <div className={styles.colorBackground} />
+    </div>
+  )
+}
+
+export const AudioRecorder = ({
+  recording,
+  setRecording,
+  saveVoiceAsDraft,
+  onError,
+}: {
+  recording: boolean
+  setRecording: (f: boolean) => void
+  saveVoiceAsDraft: (blob: Blob) => void
+  onError: (error: AudioRecorderError) => void
+}) => {
+  const tx = useTranslationFunction()
+  const [volume, setVolume] = useState(0)
+  const recorder = useRef<MicRecorder | null>(null)
+
+  const onRecordingStart = () => {
+    if (!recorder.current) {
+      recorder.current = new MicRecorder(setVolume, {
+        bitRate: 128,
+      })
+    }
+    setRecording(true)
+    recorder.current
+      ?.start()
+      .then(() => {
+        window.setTimeout(() => {
+          if (!recorder.current?.audioSignalDetected) {
+            onError(new AudioRecorderError('No input', AudioErrorType.NO_INPUT))
+            onRecordingCancel()
+          }
+        }, 1000)
+      })
+      .catch((err: any) => {
+        console.error(err)
+        onError(new AudioRecorderError(err.message))
+        setRecording(false)
+      })
+  }
+  const onRecordingStop = () => {
+    setRecording(false)
+    recorder.current
+      ?.stop()
+      .getMp3()
+      .then(([buffer, _blob]) => {
+        saveVoiceAsDraft(new Blob(buffer, { type: 'audio/mp3' }))
+      })
+      .catch((err: any) => {
+        console.error(err)
+        onError(new AudioRecorderError(err.message))
+      })
+  }
+
+  const onRecordingCancel = () => {
```

Comment:
If you cancel recording, the composer textarea becomes disabled.

Reply:
It's related again to the loadingDraft in state. I fixed that with commit [48a17](https://github.com/deltachat/deltachat-desktop/pull/4920/commits/48a1709936cfbad47379441a9e2f3a859409485f)